### ISSUE TITLE
Fused_Gromov_Loss logits confusion

### ISSUE THREAD

**Issue by Mintas** [fan23j/ICML2024-OT-CLIP#1]
You have made a great research!
Unfortunately, I am getting confused with implemetation of Fused_Gromov_Loss, as it computes logits and then tries to calculate SSM's for images and texts, but having wrong input. 

Shouldn't it be something like:
```
    def forward(self, all_image_features, all_text_features, labels):
        scores = self.fused_gromov(all_image_features, all_text_features)
        loss = F.cross_entropy(scores, labels)
        
        return {"fused_gromov_loss": loss}
```
?
https://github.com/fan23j/ICML2024-OT-CLIP/blob/c897c21ea7a69ad61f020e4774a8a0590d6702a3/training/loss.py#L155

**Comment by fan23j** [comment]
Thanks for the catch. Updated.

**Comment by Mintas** [comment]
@fan23j   Not a problem =)
similar issue here, https://github.com/fan23j/ICML2024-OT-CLIP/blob/27e3c00b510d7fbde691945d776b1c7b50d9567f/training/loss.py#L213

**Comment by Mintas** [comment]
I am currently having some problems with interchanging my vanilla CLIP loss with OT losses due to extremely small gradient values, tending to fall down to 0 and cause NaN errors...
Do you have any recommendations on this case?

**Comment by fan23j** [comment]
You can try reducing number of sinkhorn iters, or train alongside vanilla CLIP loss.
